proto/viam/app/v1/app.proto

```diff
@@ -1162,6 +1165,13 @@ message DeleteKeyRequest {
 
 message DeleteKeyResponse {}
 
+message RenameKeyRequest {
+  string id = 1;
+  string token_name = 2;
```

Comment:
can this be `new_name`? we refer to it as token name in the db but i think its fine not to here (unless there are other apis where we call it token name). also `new` makes it clear that its not the old name